Yoast/wordpress-seo
# Move frontpage form to Content Types tab and add replace var support

## Context


* Moves the form for setting the Frontpage meta title and meta description from the Social -> Facebook tab, to Search Appearance -> Content Types.
* Changes the form to the Replacevar Editor for a more coherent user-experience.

## Summary


This PR can be summarized in the following changelog entry:

* Moves the form for setting the Frontpage meta title and meta description from the Social -> Facebook tab, to Search Appearance -> Content Types.
* Changes the form to the Replacevar Editor for a more coherent user-experience.

## Relevant technical choices:

* Bit of a side-note: I've noticed that the image doesn't seem to be reflected in the source. As of writing this, the fix _has not_ yet been merged into `trunk`. 


## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

#### Testing option migration
* Ensure you have a frontpage title, description and image set under SEO -> Social -> Facebook -> Frontpage settings.
* Save your changes.
* Checkout this branch and build accordingly (`composer install`, `yarn`, `grunt build`).
* Use the test helper to change the current version to one that's lower than 16.2.
* Navigate to SEO -> Social -> Facebook.
* See that the Frontpage settings are no longer there and instead you see a notification pointing you to the Search Appearance section.
* Click in the link in the notification. You will be sent to the new location for the frontpage settings.
* See that the image, title and description that you set before, is reflected in the new location.

#### Testing the form
* Edit the image, title and description.
* See that you can now use the replace vars in your title and description.
* Save your changes.
* Navigate to the homepage of your environment and inspect the source.
* See that the changes you made to the title and description are reflected in the `og:title` and `og:description` meta fields and that the replace vars you used, are replaced with their actual values.

#### Testing disabling Open Graph meta output
* Navigate to SEO -> Social -> Facebook and set 'Add Open Graph meta data' to Disabled.
* Save your changes.
* Navigate to SEO-> Search Appearance -> Content Types and see that the Frontpage settings form is hidden.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

#### Testing option migration
* Ensure you have a frontpage title, description and image set under SEO -> Social -> Facebook -> Frontpage settings.
* Save your changes.
* Install the RC containing these changes.
* Navigate to SEO -> Social -> Facebook.
* See that the Frontpage settings are no longer there and instead you see a notification pointing you to the Search Appearance section.
* Click in the link in the notification. You will be sent to the new location for the frontpage settings.
* See that the image, title and description that you set before, is reflected in the new location.

#### Testing the form
* Edit the image, title and description.
* See that you can now use the replace vars in your title and description.
* Save your changes.
* Navigate to the homepage of your environment and inspect the source.
* See that the changes you made to the title and description are reflected in the `og:title` and `og:description` meta fields and that the replace vars you used, are replaced with their actual values.

#### Testing disabling Open Graph meta output
* Navigate to SEO -> Social -> Facebook and set 'Add Open Graph meta data' to Disabled.
* Save your changes.
* Navigate to SEO-> Search Appearance -> Content Types and see that the Frontpage settings form is hidden.

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes [P3-379]
